### PR TITLE
Visual C++ fixes and warnings

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
This PR works around a problem when building release builds with the latest version of Visual C++ (unclear if this was introduced recently or not).

It also fixes a few warnings that were introduced in a recent commit (mismatch between signed and unsigned types).

### CODE REVIEW

**Comment by MonsieurNicolas** [comment]
resolves #3067

File: src/work/test/WorkTests.cpp [hunk 1]
```diff
@@ -770,10 +770,10 @@ TEST_CASE("WorkSequence test", "[work]")
 class TestBatchWork : public BatchWork
 {
     bool mShouldFail;
-    int mTotalWorks;
+    size_t mTotalWorks;
 
   public:
-    int mCount{0};
+    size_t mCount{0};
```
Comment (marta-lokhova) [inline src/work/test/WorkTests.cpp:776]: what was the warning here? Looks like we're using ints across the board from what I see

**Comment by graydon** [comment]
r+ c8ea04e

File: src/work/test/WorkTests.cpp (hunk 1 above)
Comment (MonsieurNicolas) [reply src/work/test/WorkTests.cpp:776]: `mTotalWorks` getting initialized with a `size_t`

**Pull request merged**